WordPress/Classes/ViewRelated/Me/Views/Header/MeHeaderView.swift

```diff
@@ -0,0 +1,27 @@
+import UIKit
+
+extension MeHeaderView {
+
+    func update(with configuration: Configuration) {
+        self.gravatarEmail = configuration.gravatarEmail
+        self.username = configuration.username
+        self.displayName = configuration.displayName
+    }
+
+    struct Configuration {
+        let gravatarEmail: String?
+        let username: String
+        let displayName: String
+    }
+}
+
+extension MeHeaderView.Configuration {
+
+    init(account: WPAccount) {
+        self.init(
+            gravatarEmail: account.email,
+            username: account.username,
+            displayName: account.displayName
+        )
+    }
+}
```

Comment:
Likely a matter of personal preference, so feel free to ignore this comment, but I find it a bit confusing to have a nested typer that then has an extension outside the containing type. Maybe we could move `Configuration` outside `MeHeaderView`? or just add the initializer in the type (or initializers, if we want to preserve the init with all the params)?

Reply:
Regarding the nested type, it is inspired from some UIKit types, like `UIButton.Configuration`, but there are other types that aren't nested, like `UINavigationBarAppearance`, so I guess it's a matter of personal preference.

Would like to hear what @twstokes thinks about this and if he prefers `Configuration` to be outside `MeHeaderView` then we make it a convention moving forward.

Reply:
👋 I personally don't mind nesting types that are tightly coupled with a parent type (e.g. Error, Configuration), though I recall the defaults of [SwiftLint](https://github.com/realm/SwiftLint) complaining and [there being discussions](https://stackoverflow.com/questions/42996275/nested-types-in-swift-what-is-the-good-practice) about it.

What jumps out to me most is 

`extension MeHeaderView.Configuration {` and  `extension SupportTableViewController.Configuration {`. 

Since they have the same scope and are in the same file I'd bundle them with the type definition like @Gio2018 proposes:

> or just add the initializer in the type (or initializers, if we want to preserve the init with all the params)?

> then we make it a convention moving forward.

That'd be great to land on one!

Reply:
> so that the default init isn't removed.

These changes make sense, but I wanted to ask: Is the default init needed somewhere in the codebase currently?

Reply:
For now, the default init is only used by the convenience init `init(account: WPAccount)`. Having the default init can also help us mock `MeHeaderViewConfiguration` in case we want to write unit tests.